libraries/matrix/impl/src/main/kotlin/io/element/android/libraries/matrix/impl/room/RustMatrixRoom.kt

```diff
@@ -676,41 +676,53 @@ class RustMatrixRoom(
         return liveTimeline.sendVoiceMessage(file, audioInfo, waveform, progressCallback)
     }
 
-    override suspend fun typingNotice(isTyping: Boolean) = runCatching {
-        innerRoom.typingNotice(isTyping)
+    override suspend fun typingNotice(isTyping: Boolean) = withContext(roomDispatcher) {
+        runCatching {
+            innerRoom.typingNotice(isTyping)
+        }
     }
 
     override suspend fun generateWidgetWebViewUrl(
         widgetSettings: MatrixWidgetSettings,
         clientId: String,
         languageTag: String?,
         theme: String?,
-    ) = runCatching {
-        widgetSettings.generateWidgetWebViewUrl(innerRoom, clientId, languageTag, theme)
+    ) = withContext(roomDispatcher) {
+        runCatching {
+            widgetSettings.generateWidgetWebViewUrl(innerRoom, clientId, languageTag, theme)
+        }
     }
 
-    override fun getWidgetDriver(widgetSettings: MatrixWidgetSettings): Result<MatrixWidgetDriver> = runCatching {
-        RustWidgetDriver(
-            widgetSettings = widgetSettings,
-            room = innerRoom,
-            widgetCapabilitiesProvider = object : WidgetCapabilitiesProvider {
-                override fun acquireCapabilities(capabilities: WidgetCapabilities): WidgetCapabilities {
-                    return getElementCallRequiredPermissions(sessionId.value, deviceId.value)
-                }
-            },
-        )
+    override fun getWidgetDriver(widgetSettings: MatrixWidgetSettings): Result<MatrixWidgetDriver> {
+        return runCatching {
+            RustWidgetDriver(
+                widgetSettings = widgetSettings,
+                room = innerRoom,
+                widgetCapabilitiesProvider = object : WidgetCapabilitiesProvider {
+                    override fun acquireCapabilities(capabilities: WidgetCapabilities): WidgetCapabilities {
+                        return getElementCallRequiredPermissions(sessionId.value, deviceId.value)
+                    }
+                },
+            )
+        }
```

Comment:
There is no change here, just to confirm.

Reply:
Nope, the method is not suspending